src/openforms/prefill/contrib/objects_api/plugin.py

```diff
@@ -20,6 +26,30 @@
 class ObjectsAPIPrefill(BasePlugin):
     verbose_name = _("Objects API")
 
+    @classmethod
+    def get_prefill_values_from_mappings(
+        cls,
+        submission: Submission,
+        form_variable: FormVariable,
+    ) -> dict[str, str]:
+        variables_mappings = form_variable.prefill_options.get("variables_mapping")
+        config_group_id = form_variable.prefill_options.get("objects_api_group")
+        config_group = ObjectsAPIGroupConfig.objects.get(id=config_group_id)
```

Comment:
this can have quite some runtime crashes if the data in `prefill_options` is not as expected, especially when using `.get` where you really expect the config option to always be provided.

Take a look at `openforms.registrations.base.BasePlugin`, particularly with regard to the `Generic[OptionsT]` - you can pass a `TypedDict` what the configuration options dict is supposed to look like. A concrete example is `openforms.registrations.contrib.zgw_apis.plugin.ZGWRegistration.register_submission`, where you have a properly typed `options` argument.

I think applying a similar interface also makes more sense - why should the plugin do the lookup of `prefill_options` when the generic tooling can already provide that? So as a method signature I'd rather expect:

```py
class Options(TypedDict):
    pass


class BasePlugin[OptionsT: Options](AbstractBasePlugin):
    def get_prefill_values_from_options(
        cls, submission: Submission, options: OptionsT,
    ) -> Mapping[str, str]: ...
```

This defines the `BasePlugin` to have a type var `OptionsT` which is bound by `Options` (basically saying it *must* be a dict).

Your concrete plugin would then have:

```py
class VariableMapping(TypedDict):
    variable_key: str
    target_path: list[str]

class ObjectsAPIOptions(TypedDict):
    objects_api_group: ObjectsAPIGroupConfig
    variables_mapping: list[VariableMapping]


class Plugin(BasePlugin[ObjectsAPIOptions]):
    ...
```

The beauty here is that when you specify the serializer on the plugin class too, that you can use that already turn the JSON options into their appropriate python/django objects, [just like the registration plugins](https://github.com/open-formulieren/open-forms/blob/d0a504f523def6226ac1433a30f315dfb8523642/src/openforms/registrations/tasks.py#L277).